src/afrolabs/components/kafka/utilities.clj

```diff
@@ -130,11 +145,21 @@
       (ig/halt! system)
       (info "System done shutting down.")
 
+      (when stream-ch
+        (csp/close! stream-ch))
```

Comment:
This is a lot like `who must free this memory`... But somehow the channel must be closed so we know we are done loading data.